apps/er-coap/er-coap.c

```diff
@@ -123,7 +123,7 @@ coap_set_option_header(unsigned int delta, size_t length, uint8_t *buffer)
     } else if(*x > 12) {
       buffer[++written] = (*x - 13);
     }
-  } while(x != &length && (x = &length));
+  } while(x != (unsigned int*)&length && (x = (unsigned int*)&length));
 
```

Comment:
Well I'm not sure. I wonder whether `delta` and `length` will ever be more than 2^16−269 = 65267.